precise/iaf_psc_exp_ps_time_reversal.h

```diff
@@ -0,0 +1,589 @@
+/*
+ *  iaf_psc_exp_ps_time_reversal.h
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+
+#ifndef IAF_PSC_EXP_PS_TIME_REVERSAL_H
+#define IAF_PSC_EXP_PS_TIME_REVERSAL_H
+
+#include "config.h"
+
+#include "archiving_node.h"
+#include "nest_types.h"
+#include "event.h"
+#include "ring_buffer.h"
+#include "slice_ring_buffer.h"
+#include "connection.h"
+#include "universal_data_logger.h"
+#include "stopwatch.h"
+#include "arraydatum.h"
+
+#include <vector>
+
+/*BeginDocumentation
+Name: iaf_psc_exp_ps_time_reversal - Leaky integrate-and-fire neuron
```

Comment:
`..._time_reversal` seems like a strange name. If I understand correctly, it relates to the mathematical technique used to detect spikes. It would be more useful to the common user to provide a name that describes how the model behaves.